cluster/src/main/java/org/apache/iotdb/cluster/server/member/RaftMember.java

```diff
@@ -529,12 +531,15 @@ public long appendEntries(AppendEntriesRequest request) throws UnknownLogTypeExc
 
     long response;
     List<Log> logs = new ArrayList<>();
+    int logByteSize = 0;
     long startTime = Timer.Statistic.RAFT_RECEIVER_LOG_PARSE.getOperationStartTime();
     for (ByteBuffer buffer : request.getEntries()) {
       buffer.mark();
       Log log;
+      logByteSize = buffer.array().length;
```

Comment:
ByteBuffers from thrift may only be a slice of a larger array, so `array.length` may not refer to its real size. Use `buffer.limit() - buffer.pos()` instead.

Reply:
So, the buffer must be in read state.